### PR TITLE
Adding FCC-hh v05 production code

### PR DESCRIPTION
**Pull request by bistapf**
This PR adds all the code and config used for the v05 (and an intital tester version of v06) productions of FCC-hh samples. Specifically:

- The respective production tags and new samples are added to the `param_FCChh.py`. A customised `key4hep` release was used for the v05 production, as specified in the config. Test version for v06 currently relies on nightlies, will be made official and updated once edm4hep v1 is included in a key4hep release. The new, different FCC-hh Delphes cards for scenario I and II necessitate some changes of names in the further input files to read, these are added to the respective `run` scripts as well, hopefully not interfering with the FCC-ee code.
-  Option `--edm4hep_output_config` to run with a different output configuration file than the default hardcoded path is added to `send_lhep8` and `send_p8`. 
-  Code to retrieve and propagate the event generator weights/sum of weights for the metadata is added in `checker_yaml.py`. The event weight is stored in the `EventHeader` collection of edm4hep files produced with k4SimDelphes [v00-05](https://github.com/key4hep/k4SimDelphes/releases/tag/v00-05) or later. An optional flag can be set in the `param` config in order to use the proper sum of weights. If this is not set the behaviour should be unchanged to before. 
- MadGraph configuration files for additional samples generated for the FCC-hh analyses are added in `mg5/examples`.

Resolved WIP items:
- ~~Some FCC-hh specific instructions are added in `FCChh_Instructions/Readme.md`.~~ -> Removed, will remain on frozen [production branch for fcc_v05](https://github.com/bistapf/EventProducer/tree/fcc_v05_prod)
- ~~An additional `param_FCChh_scenarioII.py` is used to accomodate the differences between using scenario I vs scenario II in a quick and not so nice way.~~ -> From v06 FCC-hh production will switch to using the detector flag properly instead. Legacy code for the v05 production with two params remains on frozen [production branch for fcc_v05](https://github.com/bistapf/EventProducer/tree/fcc_v05_prod), but the changes made here should also be backwards compatible. Main difference is the file system structure: with detector tag scenarios I and II are put in subfolders of the production tag directory, while for v05 they were separate top level directories. 
- ~~Some additional production tags for processing some events with pile-up overlay are included, those however do not constitute a proper official production and don't need to be tracked by the events database e.g.~~ -> Removed, will remain on frozen [production branch for fcc_v05](https://github.com/bistapf/EventProducer/tree/fcc_v05_prod)
- The script to run LHE event generation in MadGraph, `submitMG.sh` is updated to run on centos7 and use an older version to be compatible with previous productions that were extended upon. ~~This is already outdated by now again, should be adapted to EL9.~~ -> Option `--centos7` is added for running LHE production with MG standalone, to support legcay running of this production for now. Looking into switching to Dirac submission with better official support for this usecase.

### CODE REVIEW

File: bin/run.py [hunk 1]
```diff
@@ -73,9 +76,12 @@
     args, _ = parser.parse_known_args()
     sendOpt = args.type
 
+    if args.customEDM4HEPOutput and not args.reco and not args.type=="lhep8":
+        parser.error("Option --customEDM4HEPOutput only works for producing edm4hep output, so if -reco and --type lhep8 options are set.")
```
[the remark this answers is not in the document]
Comment (bistapf) [reply bin/run.py]: Indeed also works for running p8 without LHE, added.

File: bin/submitMG.sh [hunk 2]
```diff
@@ -22,13 +33,17 @@ cd job
 echo "copying madgraph tarball ..."
 #cp -r /eos/experiment/fcc/hh/utils/generators/MG5_aMC_v2.6.1.tar.gz .
 #cp -r /eos/experiment/fcc/hh/utils/generators/MG5_aMC_v2.6.7.tar.gz .
-cp -r /eos/experiment/fcc/hh/utils/generators/MG5_aMC_v2.8.2.tar.gz .
+# cp -r /eos/experiment/fcc/hh/utils/generators/MG5_aMC_v2.8.2.tar.gz .
+cp -r /afs/cern.ch/work/b/bistapf/MG5_aMC_v2.5.5.tar.gz .
```
Comment (kjvbrt) [inline bin/submitMG.sh:37]: I copied `MG5_aMC_v2.5.5.tar.gz` to `/eos/experiment/fcc/hh/utils/generators` so it is in some global space. Also can you not use `-r` --- you are copying just one file.

File: bin/run.py [hunk 3]
```diff
@@ -73,9 +76,15 @@
     args, _ = parser.parse_known_args()
     sendOpt = args.type
 
+    if args.customEDM4HEPOutput and not args.reco and not (args.type=="lhep8" or args.type=="p8"):
+        parser.error("Option --customEDM4HEPOutput only works for producing edm4hep output, so if -reco and --type lhep8 or p8 options are set.")
+
```
Comment (kjvbrt) [inline bin/run.py:81]: `--reco`

File: bin/send_lhe.py [hunk 4]
```diff
@@ -95,16 +95,32 @@ def send(self):
             frun.write('cd job%s_%s\n'%(uid,self.process))
             frun.write('export EOS_MGM_URL=\"root://eospublic.cern.ch\"\n')
             frun.write('source %s\n'%(self.para.defaultstack))
-            frun.write('source /cvmfs/sft.cern.ch/lcg/views/LCG_97a_FCC_4/x86_64-centos7-gcc8-opt/setup.sh \n')
+            # frun.write('source /cvmfs/sft.cern.ch/lcg/views/LCG_97a_FCC_4/x86_64-centos7-gcc8-opt/setup.sh \n') #setting this release up clashes with defaultstack setup w.r.t python version, causing runtime errors -> removed. what is it needed for?
```
Comment (kjvbrt) [inline bin/send_lhe.py:98]: Some packages may have been missing from older stacks. Also, compiler may have been too old on the centos7.

**Pull request merged**